ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -139,8 +160,9 @@ public Object getStructFieldData(final Object data, final StructField fieldRef)
     if (data instanceof ArrayWritable) {
       final ArrayWritable arr = (ArrayWritable) data;
       final StructFieldImpl structField = (StructFieldImpl) fieldRef;
-      if (structField.getIndex() < arr.get().length) {
-        return arr.get()[structField.getIndex()];
+      int index = isRoot ? structField.getIndex() : structField.adjustedIndex;
```

Comment:
Yes, I mean adjustedIndex -> readIndex.

> 1. column is not pruned: use original index
2. column is pruned but is we are looking at the root level type: use original index
3. column is pruned and we are looking at the nested level type: use adjusted index

OK, let's keep it as ```adjustedIndex``` and use ```isRoot``` for the tag.